src/main/java/com/yelp/nrtsearch/server/luceneserver/doc/LoadedDocValues.java

```diff
@@ -354,14 +355,19 @@ public ObjectJsonDocValues(BinaryDocValues docValues) {
     public void setDocId(int docID) throws IOException {
       if (docValues.advanceExact(docID)) {
         String jsonString = STRING_DECODER.apply(docValues.binaryValue());
-        value = gson.fromJson(jsonString, List.class);
+        ListValue.Builder builder = ListValue.newBuilder();
+        JsonFormat.parser().merge(jsonString, builder);
+        value =
+            builder.getValuesList().stream()
+                .map(e -> e.getStructValue())
+                .collect(Collectors.toList());
       } else {
         value = null;
       }
     }
 
     @Override
-    public Map<String, Object> get(int index) {
+    public Struct get(int index) {
```

Comment:
Are the custom scorers using this going to be ok with using a `Struct`?

Reply:
This is object field. It should not be used by any scorer. For scoring, we will replace with maptype. Object is only used for retrieval